### PR TITLE
[WFLY-6350][WFLY-6816] Make standard configs and persisted configs match xsds

### PR DESCRIPTION
**Pull request by bstansberry**
Replaces #8773 

@stuartwdouglas @jmesnil FYI

### CODE REVIEW

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemParser_1_0.java [hunk 1]
```diff
@@ -457,8 +457,8 @@
                                                         ConnectionFactoryAttributes.Common.DUPS_OK_BATCH_SIZE,
                                                         ConnectionFactoryAttributes.Common.TRANSACTION_BATCH_SIZE,
                                                         ConnectionFactoryAttributes.Common.BLOCK_ON_ACKNOWLEDGE,
-                                                        ConnectionFactoryAttributes.Common.BLOCK_ON_DURABLE_SEND,
                                                         ConnectionFactoryAttributes.Common.BLOCK_ON_NON_DURABLE_SEND,
+                                                        ConnectionFactoryAttributes.Common.BLOCK_ON_DURABLE_SEND,
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemParser_1_0.java]: this change is wrong. In the XSD, block-on-durable-send is before block-on-non-durable-send

**Comment by jmesnil** [comment]
@bstansberry messaging-activemq changes looks good except for the block-on-durable-send bit.

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemParser_1_0.java (hunk 1 above)
Comment (bstansberry) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemParser_1_0.java]: @jmesnil different parts of the xsd (legacy-connection-factory vs the attribute group used for the other factory elements. The attribute group one has non-durable first:

https://github.com/wildfly/wildfly/blob/master/messaging-activemq/src/main/resources/schema/wildfly-messaging-activemq_1_0.xsd#L704

This reminded me though that I meant to check legacy-connection-factory and forgot. So I just did and added another commit for that one.

**Comment by jmesnil** [comment]
all looking good for the messaging-activemq subsystem, thanks for looking at it

**Comment by bstansberry** [comment]
Thanks, @rhusar for paying attention to this topic and doing the initial work. It's very helpful in general and this time surfaced a bug in core.

**Comment by rhusar** [comment]
@bstansberry Perfect, thanks for picking this up and doing this all the way.

**Pull request merged**